### PR TITLE
fix: currencies pallet changes

### LINKED ISSUE #635: Implement currency pallet changes
Currency pallet has been deprecated https://github.com/paritytech/substrate/pull/12951
calls to `currency::transfer` should be replaced by calls to `balances::transfer_allow_death` before june or it may break

note : need to maintain compatibility with chain that use this pallet but didn't upgrade yet
should be able to check if `transfer_allow_death` exists as a `call` in pallet `balances`

@chidg not sure how to triage this one

### PR DESCRIPTION
**Pull request by alecdwm**
Closes https://github.com/TalismanSociety/talisman/issues/635

Fixes for:

- [x] native balances
- [ ] orml pallet balances
- [ ] assets pallet balances
- [ ] eqBalances pallet balances

### CODE REVIEW

**Comment by alecdwm** [comment]
Should work in theory, but I'd like to test it out - maybe once Rococo merges https://github.com/paritytech/polkadot-sdk/issues/148

**Comment by chidg** [comment]
Tested as follows:
Westend has fully migrated to `transfer_allow_death`, and it works as intended using that method https://westend.subscan.io/tx/0xb24c5d369ad418c202e3deed648df075e39443ae4598988c829d096150be4b3c
Polkadot has deprecated the `transfer` method, but it still exists, so Talisman will continue to use that method until it no longer exists: https://polkadot.subscan.io/tx/0x44cc2abd63211a0e1934b10e07bbb2fae46f97908c1b2a9e3441bd4b7774b1e0

**Comment by chidg** [comment]
@alecdwm it's your PR so I can't pick you as reviewer ;). Review my changes please?

File: packages/balances-substrate-native/src/SubstrateNativeModule.ts [hunk 1]
```diff
@@ -138,6 +139,11 @@ declare module "@talismn/balances/plugins" {
   }
 }
 
+export type BalancesCommonTransferMethods = "transferKeepAlive" | "transferAll"
+export type BalancesTransferMethods = "transferAllowDeath" | BalancesCommonTransferMethods
+export type BalancesLegacyTransferMethods = "transfer" | BalancesCommonTransferMethods
+export type BalancesAllTransferMethods = BalancesLegacyTransferMethods | BalancesTransferMethods
```
Comment (alecdwm) [inline packages/balances-substrate-native/src/SubstrateNativeModule.ts:145]: Awesome 😀

**Review by alecdwm — COMMENTED** [review]
LGTM! (but I can't hit the `Approve` button because I created the inital PR 😆)

**Review by chidg — APPROVED** [review]
LGTM!

**Pull request merged**